### DISCUSSION TITLE
Kata 2 - Parte final "Usar un paquete Instalado"

### DISCUSSION THREAD

**Discussion by FernandoJiG** [LaunchX-InnovaccionVirtual/CursoIntroPython#43] [Q&A] [answered]
Hola buenas tardes, ya pude realizar el inicio de la kata pero al final en donde tenemos que usar código no sé donde ponerlo.
De la kata llegué hasta esta parte
![image](https://user-images.githubusercontent.com/99059904/153267322-6913c6fe-81c0-4c98-9db7-2995baf09d8e.png)
Pero ahora no sé en que parte poner el código final de "Usar un paquete instalado"
Lo probé en Visual Studio Code y así lo hice pero no sé si tenia que ser en el Entorno virtual :(

**Comment by Soyt0ny** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/99059703/153268614-c21c2295-aab0-41db-a60a-bd223440583b.png)

En lo personal puse la carpeta src en el la carpeta env y adentro de la carpeta src cree un arhivo .py desde vscode con el codigo que viene y lo guarde en esa carpeta para luego ejecutarlo desde el env de la kata
![image](https://user-images.githubusercontent.com/99059703/153268838-44603d85-0842-4b7f-9061-2ce7c0bf9917.png)

**Reply by FernandoJiG** [reply]
Suena bien, creo que sí es así.
Oye y como puedo ejecutar el .py desde el env?

**Reply by jdcarrillo123** [reply]
Lo que dice Claudio es correcto.
Pero en el ejercicio de la Kata "Ejercicio - Crear un paquete" no pide esa parte.

**Reply by RocioGH** [reply]
Hola, disculpa seguí tu consejo porque también estoy batallando con esa parte pero me sale el siguiente error:
![Screenshot_5](https://user-images.githubusercontent.com/99061704/153331089-8525ec9e-a28b-4597-9162-437429faf917.png)

Sabes a que se deba??

**Reply by KevinVillanuev4** [reply]
Python no ocupa ; para separ lineas ocupa espacios o tabulaciones por eso es el error basta con quitarle los espacios al principio de cada linea

from datetime import *
from dateutil.relativedelta import *
now = datetime.now()
print(now)

now = now + relativedelta(months=1, weeks=1, hour=10)

print(now)

**Reply by ChanLuis** [reply]
Oye bro, la kata del modulo 2 es subir la carpeta o solo un archivo? y cual sería?

**Reply by brauliorguez** [reply]
@ChanLuis  por lo que entendi es subir capturas mostrando el proceso de como creaste el Entorno virtual y los pasos siguientes, y otra cosas que vi es que algunos sugerían hacer un notebook e ir insertando las imagenes ahi o un PDF. Al igual creo no hay problema si subes tambien las carpetas, es evidencia de tu trabajo.

**Reply by stariluz** [reply]
Amigos creo que no es tan recomendable la forma en que lo realizo @ClaudioAGarcia (desde lo que puedo decir en base a mi corta experiencia). El directorio env no es algo que se suela subir a un repositorio, si no que cada persona que colabora en el proyecto tiene su propio env con las librerias correspondientes instaladas (solamente se guarda un archivo llamado requirements.txt que indica las librerias y su versión, para que cada colaborador instale las correctas). Poner el código de nuestra "aplicacion" dentro del env, traería problemas a la hora que otros traten de colaborar con el proyecto en su máquina. Según me enseñaron  las carpetas deben estar separadas como lo señalan en la explicación de la kata:
```
/project
    /env 
        /bin
        /include
        /lib
        . . .
    /src
        app.py 
        . . .
```
Esto tengo entendido yo, la verdad si hay alguien con más experiencia en el asunto, agradecería la respectiva corrección o aclaración. Gracias

**Reply by Soyt0ny** [reply]
Tienes razón fallo mío no había visto el acomodo en la actividad gracias @Adrigondo
